src/libstd/sys_common/gnu/libbacktrace.rs

```diff
@@ -135,7 +151,7 @@ pub fn print(w: &mut Write, idx: isize, addr: *mut libc::c_void,
 
     // backtrace errors are currently swept under the rug, only I/O
     // errors are reported
-    let state = unsafe { init_state() };
+    let (state, _file) = unsafe { init_state() };
```

Comment:
`_file` needs to be unlocked (dropped) only if we got success from `backtrace_syminfo`.
Otherwise libbacktrace will try to read it again in next backtrace requests in unlocked state.
On failure the file needs to be either leaked or stored to be unlocked later.